packages/react-frontend/src/routes/Album.jsx

```diff
@@ -20,6 +19,40 @@ async function fetchArtistBySpotifyId(spotifyId) {
   }
 }
 
+// Function to fetch reviews and calculate floored average rating
+async function fetchReviewsAndCalculateRatings() {
+  try {
+    const response = await fetch("http://localhost:8000/reviews");
+    if (!response.ok) {
+      throw new Error("Failed to fetch reviews");
+    }
+    const reviews = await response.json();
+    const albumRatings = {};
+
+    // Calculate the total rating and count for each album
+    reviews.forEach((review) => {
+      const { album_id, rating } = review;
+      if (!albumRatings[album_id]) {
+        albumRatings[album_id] = { totalRating: 0, count: 0 };
+      }
+      albumRatings[album_id].totalRating += rating;
+      albumRatings[album_id].count += 1;
+    });
+
+    // Calculate the floored average rating for each album
+    const averageRatings = {};
+    for (const albumId in albumRatings) {
+      const { totalRating, count } = albumRatings[albumId];
+      averageRatings[albumId] = Math.floor(totalRating / count);
+    }
```

Comment:
I understand what you're going for here and how it works. However it seems a bit inefficient and maybe contributing to the slowness of the filtering.

Reply:
Hmmm, I agree. Do you have any ideas on how to make the code more efficient? Maybe another way to fetch the data or what?

Reply:
I agree that there may be a more efficient way to do this. One thing that comes to mind is storing the average rating for each album in the JSON data, then updating that value with POST api each time a new review is submitted.

Reply:
Ooooo, that sounds like that could work. I'll look into it, thank you for the suggestion!

Reply:
I think because the filter works out, I don't think I'm going to change the code. It's also faster after I changed the album schema.